python/strings_udf/strings_udf/__init__.py

```diff
@@ -87,6 +88,29 @@ def _get_ptx_file():
         return regular_result[1]
 
 
+# Maximum size of a string column is 2gb
+STRINGS_UDF_DEFAULT_HEAP_SIZE = 2**31
+STRINGS_UDF_HEAP_SIZE = os.environ.get(
+    "STRINGS_UDF_HEAP_SIZE", STRINGS_UDF_DEFAULT_HEAP_SIZE
+)
+heap_size = 0
+
+
+def set_malloc_heap_size(size=STRINGS_UDF_HEAP_SIZE):
```

Comment:
Why should this have a default, rather than reading the env var when it is lazily allocated?

I’m not convinced `STRINGS_UDF_HEAP_SIZE` needs to be defined.

Reply:
The pool is _very_ small if unset - something like 8mb I believe. Perhaps it'd be better if it said something like `_STRINGS_UDF_INITIAL_HEAP_SIZE`?

Reply:
I think @bdice is just asking why there's a default function parameter at all if this is set by an environment variable. This sort of relates to the other question about why we call this lazily rather than at initialization. If it needs to be done lazily for a good reason, then perhaps a better alternative would be to have the default `size=None` and read the environment variable inside the function if `size==None` instead of storing it in a module-scope function. That shouldn't lead to repeated reads since after the first invocation with the current state of the code `heap_size` will always be equal to `size` so it won't go into the conditional.

Reply:
The lazy initialization is kind of a placeholder in case we have an algorithm that can help guess the size based on the input size for example. I cannot speak to the env var access since I'm not familiar with the scoping here as it relates to udf kernel launches. Ideally if it could be set before each kernel launch? Though I understand the env var may not have changed.

Reply:
+1 on what @davidwendt mentioned about initialization. For the logic discussion, what do you think of the changes in https://github.com/rapidsai/cudf/pull/12094/commits/af7cd9f85b2afa0a6b92f53e4abffbb538c9d02f ?